### PR TITLE
Refactors  plot_ref_sld to plot_ref_sld_helper,

### PR DESCRIPTION
**Pull request by StephenNneji**
- Refactors plot_ref_sld to plot_ref_sld_helper
- Creates a new plot_ref_sld function that accepts problem and result
-  Ensures plots are extended similar to Matlab plot

### CODE REVIEW

File: RAT/utils/plotting.py [hunk 1]
```diff
@@ -154,25 +153,29 @@ def plot_ref_sld(data: PlotEventData, fig: Figure = None, delay: bool = True):
...
-                          [row[1] for row in new],
-                          color=color,
-                          linewidth=1)
+            layers = data.resampledLayers[i][0]
+            for j in range(len(data.resampledLayers[i])):
+                layer = data.resampledLayers[i][j]
+                if layers.shape[1] == 4:
+                    layer = np.delete(layer, 2, 1)
+                new_profile = makeSLDProfileXY(layers[0, 1],      # Bulk In
+                                               layers[-1, 1],     # Bulk Out
```
Comment (DrPaulSharp) [inline RAT/utils/plotting.py:169]: The code you sent me to address this had this line as: "layers[-1, 2]" - could you double check which it should be for me please.

File: RAT/utils/plotting.py [hunk 2]
```diff
@@ -17,9 +18,9 @@ def __init__(self, row: int = 1, col: int = 2):
 
         Parameters
         ----------
-        row : int
+        row : int, default: 1
               The number of rows in subplot
-        col : int
+        col : int, default: 2
```
Comment (DrPaulSharp) [inline RAT/utils/plotting.py:23]: Should the default be 2 here for the general figure routine? Do you think it makes more sense to have the default as 1 here and set two columns in plot_ref_sld?

File: RAT/utils/plotting.py [hunk 3]
```diff
@@ -191,3 +194,33 @@ def plot_ref_sld(data: PlotEventData, fig: Figure = None, delay: bool = True):
...
+
+
+def plot_ref_sld(problem, results, block: bool = False):
+    """
+    Plots the reflectivity and SLD profiles.
+
+    Parameters
+    ----------
+    problem : Project
+              An instance of the Project class
```
Comment (DrPaulSharp) [inline RAT/utils/plotting.py:206]: As written, this should be an instance of "ProblemDefinition", not the project class.

File: RAT/utils/plotting.py (hunk 2 above)
Comment (StephenNneji) [reply RAT/utils/plotting.py:23]: I will set default to 1

**Review by DrPaulSharp — APPROVED** [review]

**Pull request merged**